src/OpenTelemetry/.publicApi/net462/PublicAPI.Shipped.txt

```diff
@@ -18,8 +18,8 @@ OpenTelemetry.Metrics.MetricPoint.GetHistogramBuckets() -> OpenTelemetry.Metrics
 ~OpenTelemetry.Metrics.MetricStreamConfiguration.TagKeys.get -> string[]
 ~OpenTelemetry.Metrics.MetricStreamConfiguration.TagKeys.set -> void
 ~OpenTelemetry.Metrics.PeriodicExportingMetricReader.PeriodicExportingMetricReader(OpenTelemetry.BaseExporter<OpenTelemetry.Metrics.Metric> exporter, int exportIntervalMilliseconds = 60000, int exportTimeoutMilliseconds = 30000) -> void
-~override OpenTelemetry.Metrics.MeterProviderBuilderBase.AddInstrumentation<TInstrumentation>(System.Func<TInstrumentation>! instrumentationFactory) -> OpenTelemetry.Metrics.MeterProviderBuilder!
-~override OpenTelemetry.Trace.TracerProviderBuilderBase.AddInstrumentation<TInstrumentation>(System.Func<TInstrumentation>! instrumentationFactory) -> OpenTelemetry.Trace.TracerProviderBuilder!
+override OpenTelemetry.Metrics.MeterProviderBuilderBase.AddInstrumentation<TInstrumentation>(System.Func<TInstrumentation!>! instrumentationFactory) -> OpenTelemetry.Metrics.MeterProviderBuilder!
+override OpenTelemetry.Trace.TracerProviderBuilderBase.AddInstrumentation<TInstrumentation>(System.Func<TInstrumentation!>! instrumentationFactory) -> OpenTelemetry.Trace.TracerProviderBuilder!
```

Comment:
These were updated because of the changes to OpenTelemetry.Api.

`MeterProviderBuilder` and `TracerProviderBuilder` both have an `AddInstrumentation()` method. The `TInstrumentation` is now defined as not-nullable which affects the `Func<TInstrumentation>` parameter of the methods in the OpenTelemetry project.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/410532ae491fa02d96c1cf0325e7baa0703d9fb4/src/OpenTelemetry/Metrics/Builder/MeterProviderBuilderBase.cs#L69-L79

https://github.com/open-telemetry/opentelemetry-dotnet/blob/410532ae491fa02d96c1cf0325e7baa0703d9fb4/src/OpenTelemetry.Api/Metrics/MeterProviderBuilder.cs#L37-L39

Reply:
This is breaking API compat: https://github.com/open-telemetry/opentelemetry-dotnet/actions/runs/4625983584/jobs/8182256455?pr=4339